pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -39,15 +39,11 @@ library BasePoolMath {
         // bpt = bptTotalSupply                                                            //
         ************************************************************************************/
 
-        // Since we're computing amounts in, we round up overall. This means rounding up on both the
-        // multiplication and division.
-
-        uint256 bptRatio = bptAmountOut.divUp(bptTotalSupply);
-
         // Create a new array to hold the amounts of each token to be deposited.
         amountsIn = new uint256[](balances.length);
         for (uint256 i = 0; i < balances.length; ++i) {
-            amountsIn[i] = balances[i].mulUp(bptRatio);
+            // Since we multiply and divide we don't need to use FP math.
+            amountsIn[i] = (balances[i] * bptAmountOut) / bptTotalSupply;
```

Comment:
Do we need to add 1 to preserve the rounding direction? `(balances[i] * bptAmountOut + 1) / bptTotalSupply`

Reply:
yeah in this case it might still make sense. Adjusted.

Reply:
> `(balances[i] * bptAmountOut + 1) / bptTotalSupply`

I think that's not the formula though. Probably better to do a raw multiplication and then divUp instead.